module/VuFind/src/VuFind/Auth/Manager.php

```diff
@@ -193,15 +194,37 @@ public function supportsCreation($authMethod = null)
     /**
      * Does the current configuration support password recovery?
      *
-     * @param string $authMethod optional; check this auth method rather than
-     *  the one in config file
+     * @param string $authMethod optional; check this auth method rather than the one in config file
+     * @param string $target     Login target (only for MultiILS)
      *
      * @return bool
      */
-    public function supportsRecovery($authMethod = null)
+    public function supportsRecovery($authMethod = null, $target = '')
     {
         return ($this->config->Authentication->recover_password ?? false)
-            && $this->getAuth($authMethod)->supportsPasswordRecovery();
+            && $this->getAuth($authMethod)->supportsPasswordRecovery($target);
+    }
+
+    /**
+     * Get password recovery data (such as a user id or recovery token) based on form data submitted by the user.
+     *
+     * @param array $params Request params (form data)
+     *
+     * @return ?array Null if user not found, or associative array with following keys:
+     *   string email       User's email address
+     *   string username    Username (optional, for display)
+     *   string auth_method Authentication method
+     *   string target      Authentication target for methods that support target selection (e.g. MultiILS)
+     *   int    timestamp   Request timestamp (unix time)
+     *   array  details  Array of user details required for resetPassword request
```

Comment:
This row doesn't line up with the rest:

```suggestion
     *   array  details     Array of user details required for resetPassword request
```